### PR TITLE
ComponentUT header and implement

### PR DESCRIPTION
**Pull request by B-10515025**

### CODE REVIEW

**Review by harrisonyei — CHANGES_REQUESTED** [review]
I commit some changes to the comments of both files.
Other than that, LGTM

File: src/Components/ComponentUT.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    long long getWeightedSum();
+protected:
+    // Override component function
+    // Reset variable
+    void start();
+    // Override component function
+    // Return true if queue is enpty
+    bool wait();
+    // Override component function
+    // Calculation weighted sum of input data
```
Comment (firewings2515) [inline src/Components/ComponentUT.h]: // Calculate weighted sum of input data

File: src/Components/ComponentUT.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    // The number of input data
+    const int test_time = 10000;
+    long long weighted_sum = 0;
+    ComponentUT component_unit;
+    component_unit.run();
+    srand(time(NULL));
+    for (int i = 1; i <= test_time; i++) {
+        // Random generate input data
+        int input = rand();
+        // Calculation weighted sum in main thread
```
Comment (firewings2515) [inline src/Components/ComponentUT.cpp]: // Calculate weighted sum in main thread

File: src/Components/ComponentUT.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+bool ComponentUT::wait() {
+    return getSize() == 0;
+}
+
+void ComponentUT::process() {
+    std::shared_ptr<int> num;
+    num = getData();
+    weighted_sum += *num * count++;
```
Comment (firewings2515) [inline src/Components/ComponentUT.cpp]: I'm not sure, but I think using parentheses to *num is better.
(e.i. (*num))

**Review by firewings2515 — APPROVED** [review]
Fixed some grammar and format.
LGTM

**Review by AriaZhou — APPROVED** [review]

File: src/Components/ComponentUT.cpp [hunk 4]
```diff
@@ -0,0 +1,75 @@
+﻿// MIT License
```
Comment (john0312) [inline src/Components/ComponentUT.cpp]: Usually, the coding style specify tests files to be named as: something_test.cc
You can follow the coding style or have a consistent style for your project.

File: src/Components/ComponentUT.h [hunk 5]
```diff
@@ -0,0 +1,56 @@
+// MIT License
```
Comment (john0312) [inline src/Components/ComponentUT.h]: A header file is not needed for test usually, because we don't share symbols with other .cc files.

**Review by john0312 — APPROVED** [review]

File: src/Components/ComponentUT.h (hunk 1 above)
Comment (john0312) [reply src/Components/ComponentUT.h]: This is a reasonable comment.

**Pull request merged**